prov/efa/src/efa_fabric.c

```diff
@@ -853,6 +854,29 @@ int efa_fabric(struct fi_fabric_attr *attr, struct fid_fabric **fabric_fid,
 	struct efa_fabric *fab;
 	int ret = 0;
 
+	/*
+	 * Enable rdma-core fork support and huge page support. We want call
+	 * this only when the EFA provider is selected. It is safe to call this
+	 * function again if multiple EFA fabrics are opened or if the fabric
+	 * is closed and opened again.
+	 *
+	 * TODO: allow users to disable this once the fork() to check ptrace
+	 * permissions is removed.
+	 */
+	ret = setenv("RDMAV_HUGEPAGES_SAFE", "1", 1);
+	if (ret)
+		return -errno;
+
+	ret = ibv_fork_init();
```

Comment:
FWIW, we had this call in the verbs provider, but had to remove it.  It caused a performance hit, but also didn't work.  An application might access verbs through multiple middleware libraries, and there's no guarantee that the other library doesn't do something with verbs prior to this being called.  The only viable solution was to use the verbs fork environment variable.  Once that's a requirement, there's no need to call this, which also removes the performance impact when fork isn't used.

Reply:
Right, this isn't ideal from both a performance standpoint and avoiding a race with something accessing verbs before Libfabric initialization. However, we're concerned that our customers are not going to know whether they need fork support or not. We should be able to mostly avoid another middleware library accessing verbs on EC2. I put the call in EFA's fabric init as to avoid impacting other Libfabric providers here with our workaround.

Reply:
This caused a problem with initialization that was a functional problem, not a performance problem. I'll see if I can dig up the issue.

Reply:
See https://github.com/ofiwg/libfabric/pull/4973